README.md
# Tor Portable #

**Tor Portable** ( *include Tor, Vidalia, Polipo* ) package from **Tor Browser Bundle for Windows**.

- [Tor](https://www.torproject.org)
- [Vidalia](https://www.torproject.org/projects/vidalia.html.en "Vidalia is a cross-platform graphical controller for the Tor software, built using the Qt framework.")
- [Polipo](http://www.pps.univ-paris-diderot.fr/~jch/software/polipo/ "Polipo is a small and fast caching web proxy (a web cache, an HTTP proxy, a proxy server).")
- [Solipo](http://serennz.sakura.ne.jp/toybox/solipo/ "Solipo is GUI wrapper application for polipo on Windows.")
- [Tor Browser](https://www.torproject.org/projects/torbrowser.html.en "The Tor Browser Bundle lets you use Tor on Windows, Mac OS X, or Linux without needing to install any software.") ( *Firefox* )

# What is Tor? #

[Tor](https://www.torproject.org/ "Tor") is free software and an open network that helps you defend against a form of network surveillance that threatens personal freedom and privacy, confidential business activities and relationships, and state security known as traffic analysis 

# Polipo Admin #

[http://127.0.0.1:8118/polipo/](http://127.0.0.1:8118/polipo/)

# Tor Proxy Setting #

Choose one for your proxy setting.

- 127.0.0.1:8118 ( *proxy by Polipo* )
- 127.0.0.1:9050 ( *direct by Tor, Socks v5* )

You can check [Are you using Tor?](https://check.torproject.org/) now.


Docs/changelog
Tor Browser Bundle (2.3.25-2); suite=windows

  * Update Firefox to 10.0.12esr
  * Update Libevent to 2.0.21-stable
  * Update HTTPS Everywhere to 3.1.2
  * Update NoScript to 2.6.4.2

 -- Erinn Clark <erinn@torproject.org>  Fri Jan  4 11:46:07 CET 2013

Tor Browser Bundle (2.3.25-1); suite=windows

  * Update Tor to 0.2.3.25
  * Update Firefox 10.0.11esr
  * Update Vidalia to 0.2.21
  * Update NoScript to 2.6.2

 -- Erinn Clark <erinn@torproject.org>  Sun Dec  2 09:03:27 GMT 2012

Tor Browser Bundle (2.3.24-alpha-1); suite=windows

  * Update Tor to 0.2.3.24-rc
  * Update Firefox to 10.0.10esr
  * Update NoScript to 2.5.9
  * Update HTTPS Everywhere to 4.0development.2

 -- Erinn Clark <erinn@torproject.org>  Sat Oct 27 12:16:57 BST 2012

Tor Browser Bundle (2.3.23-alpha-1); suite=windows

  * Update Tor to 0.2.3.23-rc
  * Update Firefox to 10.0.9esr
  * Update HTTPS Everywhere to 4.0development.1
  * Update NoScript to 2.5.8
  * Re-enable automatic Control and SOCKS port selection on Linux and OSX

 -- Erinn Clark <erinn@torproject.org>  Mon Oct 22 16:14:05 BST 2012

Tor Browser Bundle (2.3.22-alpha-1); suite=windows

  * Update Tor to 0.2.3.22-rc

 -- Erinn Clark <erinn@torproject.org>  Tue Sep 11 19:49:08 BST 2012

Tor Browser Bundle (2.3.21-alpha-1); suite=windows

  * Update Tor to 0.2.3.21-rc
  * Update Firefox to 15.0.1
  * Update Libevent to 2.0.20-stable
  * Update Torbutton to 1.4.6.1
  * Update HTTPS Everywhere to 3.0development.6
  * Update NoScript to 2.5.4

 -- Erinn Clark <erinn@torproject.org>  Sun Sep  9 10:42:35 BST 2012

Tor Browser Bundle (2.3.20-alpha-1); suite=windows

  * Update Tor to 0.2.3.20-rc
  * Update NoScript to 2.5
  * Change the urlbar search engine to Startpage (closes: #5925)
  * Firefox patch updates:
    - Fix the Tor Browser SIGFPE crash bug (closes: #6492)
    - Add a redirect API for HTTPS-Everywhere (closes: #5477)
    - Enable WebGL (as click-to-play only) (closes: #6370)

 -- Erinn Clark <erinn@torproject.org>  Sun Aug  5 23:21:32 BST 2012

Tor Browser Bundle (2.3.19-alpha-1); suite=windows

  * Update Tor to 0.2.3.19-rc
  * Update Firefox to 14.0.1
  * Update libevent to 2.0.19-stable
  * Update OpenSSL to 1.0.1c
  * Update zlib to 1.2.7
  * Update Torbutton to 1.4.6
  * Update NoScript to 2.4.9
  * Update HTTPS Everywhere to 3.0development.5
  * Downgrade Vidalia to 0.2.20

 -- Erinn Clark <erinn@torproject.org>  Wed Jul 25 15:12:41 BST 2012

Tor Browser Bundle (2.3.12-alpha-2); suite=windows

  * Update Firefox to 11.0
  * Update NoScript to 2.3.4
  * Update HTTPS Everywhere to 3.0development.1
  * Disable HTTPS Everywhere SSL Observatory screen (closes: #5300)
  * Always build to with warnings enabled (closes: #4470)
  * Remove tor-resolve from the Windows bundle (closes: #5403)

 -- Erinn Clark <erinn@torproject.org>  Mon Mar 19 01:14:43 BRT 2012

Tor Browser Bundle (2.3.12-alpha-1); suite=windows

  * Initial release

 -- Erinn Clark <erinn@torproject.org>  Thu Mar  1 14:04:56 BRT 2012


FirefoxPortable/Data/profile/extensions/{e0204bd5-9d31-402b-a99d-a6aa8ffebdca}/chrome/content/jshooks.js
// Bug 1506 P0: These hooks are only relevant to FF 3.5 and below.

window.__HookObjects = function() {
  if (typeof(window.__tb_hooks_ran) === "boolean") {
      return false;
  }


  /* For reference/debugging only:
  if(false && window.__tb_set_uagent===true) {
      var tmp_oscpu = window.__tb_oscpu;
      var tmp_platform = window.__tb_platform;
      var tmp_productSub = window.__tb_productSub;

      try {
          if(!(window.navigator.__proto__ == null) ||
                  typeof(window.navigator.__defineGetter__) === "function") {
              var cE = window.navigator.cookieEnabled;
              var lang = window.navigator.language;
              var uA = window.navigator.userAgent;
              var v = window.navigator.vendor;
              var vS = window.navigator.vendorSub;
              var jE = window.navigator.javaEnabled;
              var pl = new window.Array();
              var mT = new window.Object();
              //var pl = window.navigator.plugins;
              //var mT = window.navigator.mimeTypes;

              window.navigator.__defineGetter__("appCodeName", function() { return "Mozilla";});
              window.navigator.__defineGetter__("appName", function() { return "Netscape";});
              window.navigator.__defineGetter__("appVersion", function() { return "5.0";});
              window.navigator.__defineGetter__("buildID", function() { return 0;});
              window.navigator.__defineGetter__("cookieEnabled", function() { return cE;});
              window.navigator.__defineGetter__("language", function() { return lang;});
              window.navigator.__defineGetter__("mimeTypes", function() { return mT;});
              window.navigator.__defineGetter__("onLine", function() { return true;});
              window.navigator.__defineGetter__("oscpu", function() { return tmp_oscpu;});
              window.navigator.__defineGetter__("platform", function() { return tmp_platform;});
              window.navigator.__defineGetter__("plugins", function() { return pl;});
              window.navigator.__defineGetter__("product", function() { return "Gecko";});
              window.navigator.__defineGetter__("productSub", function() { return tmp_productSub;});
              window.navigator.__defineGetter__("securityPolicy", function() { return "";});
              window.navigator.__defineGetter__("userAgent", function() { return uA;});
              window.navigator.__defineGetter__("vendor", function() { return v;});
              window.navigator.__defineGetter__("vendorSub", function() { return vS;});
              window.navigator.__defineGetter__("javaEnabled", function() { return jE;});
              window.navigator.__proto__ = null;
          }
      } catch(e) {
      }
  } */


  /* Hrmm.. Is it possible this breaks plugin install or other weird shit
     for non-windows OS's? */
  if(window.__tb_set_uagent===true) {
      var tmp_oscpu = window.__tb_oscpu;
      var tmp_platform = window.__tb_platform;
      var tmp_productSub = window.__tb_productSub;
      var tmp_locale = window.__tb_locale;

      // This is just unreasonable.. Firefox caches 
      // window.navigator.__proto__ between same-origin loads of a document. 
      // So this means when we null it out, we lose most of the navigator 
      // object for subsequent loads. I tried doing the whole-object hooks 
      // like we do for Date, screen, and history, but it seems to behave 
      // like Date upon delete, allowing unmasking for that case. Talk about 
      // rock+hard place. 
      try {
          if(!(window.navigator.__proto__ == null) ||
             typeof(window.navigator.__defineGetter__) === "function") {
              var tmpNav = new window.Object();
              for(var i in window.navigator) {
                  tmpNav[i] = window.navigator[i];
                  
                  // XPCNative objects are special for some reason. So far, 
                  // all we have are "plugins" and mimeTypes, which 
                  // are empty anyways. Disable them.
                  //if(tmpNav[i].toString().indexOf("XPCNative") != -1) {
                  if(i === "plugins" || i === "mimeTypes") {
                      tmpNav[i] = new Array();
                  }
                  (function() { // crazy scope hack to preserve i
                      var holder = i;
                      window.navigator.__defineGetter__(i, function() { return tmpNav[holder];});
                  })();
              }

              if(tmp_locale != false) {
                  window.navigator.__defineGetter__("language", function() { return tmp_locale;});
              }
              window.navigator.__defineGetter__("oscpu", function() { return tmp_oscpu;});
              window.navigator.__defineGetter__("productSub", function() { return tmp_productSub;});
              window.navigator.__defineGetter__("buildID", function() { return 0;});
              window.navigator.__proto__ = null;
          }
      } catch(e) {
      }
  }

  // No pref for this.. Should be mostly harmless..
  if(true) {
      var origHeight = window.innerHeight;
      var origWidth = window.innerWidth;
      window.__proto__.__defineGetter__("innerHeight", function() { return Math.round(origHeight/50.0)*50;});
      window.__proto__.__defineGetter__("innerWidth", function() { return Math.round(origWidth/50.0)*50;});
      window.__proto__.__defineGetter__("outerWidth", function() { return window.innerWidth;});
      window.__proto__.__defineGetter__("outerHeight", function() { return window.innerHeight;});
      window.__proto__.__defineGetter__("screenX", function() { return 0;});
      window.__proto__.__defineGetter__("screenY", function() { return 0;});
      window.__proto__.__defineGetter__("pageXOffset", function() { return 0;});
      window.__proto__.__defineGetter__("pageYOffset", function() { return 0;});

      // Wrap in anonymous function to protect scr variables just in case.
      (function () {
          // We can't define individual getters/setters for window.screen 
          // for some reason. works in html but not in these hooks.. No idea why

          var scr = new Object();
          var origScr = window.screen;
          scr.__defineGetter__("height", function() { return window.innerHeight; });
          scr.__defineGetter__("width", function() { return window.innerWidth; });

          scr.__defineGetter__("availTop", function() { return 0;});
          scr.__defineGetter__("availLeft", function() { return 0;});

          scr.__defineGetter__("top", function() { return 0;});
          scr.__defineGetter__("left", function() { return 0;});

          scr.__defineGetter__("availHeight", function() { return window.innerHeight;});
          scr.__defineGetter__("availWidth", function() { return window.innerWidth;});

          scr.__defineGetter__("colorDepth", function() { return origScr.colorDepth;});
          scr.__defineGetter__("pixelDepth", function() { return origScr.pixelDepth;});

          scr.__defineGetter__("availTop", function() { return 0;});
          scr.__defineGetter__("availLeft", function() { return 0;});

          window.__defineGetter__("screen", function() { return scr; });
          window.__defineSetter__("screen", function(a) { return; });
          window.__proto__.__defineGetter__("screen", function() { return scr; });

          // Needed for Firefox bug 418983:
          with(window) {
              var screen = scr;
          }

      })();

  }
  

  if(window.__tb_hook_date == true) { // don't feel like indenting this

  /* Timezone fix for http://gemal.dk/browserspy/css.html */
  var reparseDate = function(d, str) {
    /* Rules:
     *   - If they specify a timezone, it is converted to local
     *     time. All getter fucntions then convert properly to
     *     UTC. No mod needed.
     *   - If they specify UTC timezone, then it is converted
     *     to local time. All getter functions then convert back.
     *     No mod needed.
     *   - If they specify NO timezone, it is local time. 
     *     The UTC conversion then reveals the offset. Fix.
     */
    
    /* Step 1: Remove everything inside ()'s (they are "comments") */
    var s = str.toLowerCase();
    var re = new RegExp('\\(.*\\)', "gm");
    s = s.replace(re, "");

    /* Step 2: Look for +/-. If found, do nothing */
    if(s.indexOf("+") == -1 && s.indexOf("-") == -1) {
      /* Step 3: Look for timezone string from
       * http://lxr.mozilla.org/seamonkey/source/js/src/jsdate.c
       */
      if(s.indexOf(" gmt") == -1 && s.indexOf(" ut") == -1 &&
         s.indexOf(" est") == -1 && s.indexOf(" edt") == -1 &&
         s.indexOf(" cst") == -1 && s.indexOf(" cdt") == -1 &&
         s.indexOf(" mst") == -1 && s.indexOf(" mdt") == -1 &&
         s.indexOf(" pst") == -1 && s.indexOf(" pdt")) {
        /* No timezone specified. Adjust. */
        d.setTime(d.getTime()-(d.getTimezoneOffset()*60000));
      }
    } 
  } 
  
  var origDate = window.Date;
  var newDate = function() {
    /* DO NOT make 'd' a member! EvilCode will use it! */
    var d;
    var a = arguments;
    /* apply doesn't seem to work for constructors :( */
    if(arguments.length == 0) d=new origDate();
    if(arguments.length == 1) d=new origDate(a[0]);
    if(arguments.length == 3) d=new origDate(a[0],a[1],a[2]);
    if(arguments.length == 4) d=new origDate(a[0],a[1],a[2],a[3]);
    if(arguments.length == 5) d=new origDate(a[0],a[1],a[2],a[3],a[4]);
    if(arguments.length == 6) d=new origDate(a[0],a[1],a[2],a[3],a[4],a[5]);
    if(arguments.length == 7) d=new origDate(a[0],a[1],a[2],a[3],a[4],a[5],a[6]);
    if(arguments.length > 7) d=new origDate();

    if(arguments.length > 0) {
      if((arguments.length == 1) && typeof(a[0]) == "string") {
        reparseDate(d,a[0]);
      } else if(arguments.length > 1) { 
        /* Numerical value. No timezone given, adjust. */
        d.setTime(d.getTime()-(d.getTimezoneOffset()*60000));
      }
    }

    // XXX: the native valueOf seems to sometimes return toString and 
    // sometimes return getTime depending on context (which we can't detect)
    // It seems as though we can't win here..
    window.Date.prototype.valueOf=function(){return d.getTime()};
    window.Date.prototype.getTime=function(){return d.getTime();} /* UTC already */ 
    window.Date.prototype.getFullYear=function(){return d.getUTCFullYear();}  
    window.Date.prototype.getYear=function() {return d.getYear();}
    window.Date.prototype.getMonth=function(){return d.getUTCMonth();}
    window.Date.prototype.getDate=function() {return d.getUTCDate();}
    window.Date.prototype.getDay=function() {return d.getUTCDay();}
    window.Date.prototype.getHours=function(){return d.getUTCHours();}
    window.Date.prototype.getMinutes=function(){return d.getUTCMinutes();}
    window.Date.prototype.getSeconds=function(){return d.getUTCSeconds();}
    window.Date.prototype.getMilliseconds=function(){return d.getUTCMilliseconds();}
    window.Date.prototype.getTimezoneOffset=function(){return 0;}
 
    window.Date.prototype.setTime = 
       function(x) {return d.setTime(x);}
    window.Date.prototype.setFullYear=function(x){return d.setUTCFullYear(x);}
    window.Date.prototype.setYear=function(x){return d.setYear(x);}
    window.Date.prototype.setMonth=function(x){return d.setUTCMonth(x);}
    window.Date.prototype.setDate=function(x){return d.setUTCDate(x);}
    window.Date.prototype.setDay=function(x){return d.setUTCDay(x);}
    window.Date.prototype.setHours=function(x){return d.setUTCHours(x);}
    window.Date.prototype.setMinutes=function(x){return d.setUTCMinutes(x);}
    window.Date.prototype.setSeconds=function(x){return d.setUTCSeconds(x);}
    window.Date.prototype.setMilliseconds=
       function(x) {return d.setUTCMilliseconds(x);}
 
    window.Date.prototype.getUTCFullYear=function(){return d.getUTCFullYear();}  
    window.Date.prototype.getUTCMonth=function(){return d.getUTCMonth();}
    window.Date.prototype.getUTCDate=function() {return d.getUTCDate();}
    window.Date.prototype.getUTCDay=function() {return d.getUTCDay();}
    window.Date.prototype.getUTCHours=function(){return d.getUTCHours();}
    window.Date.prototype.getUTCMinutes=function(){return d.getUTCMinutes();}
    window.Date.prototype.getUTCSeconds=function(){return d.getUTCSeconds();}
    window.Date.prototype.getUTCMilliseconds=
       function(){return d.getUTCMilliseconds();}
     
    window.Date.prototype.setUTCFullYear=function(x){return d.setUTCFullYear(x);}
    window.Date.prototype.setUTCMonth=function(x){return d.setUTCMonth(x);}
    window.Date.prototype.setUTCDate=function(x){return d.setUTCDate(x);}
    window.Date.prototype.setUTCDay=function(x){return d.setUTCDay(x);}
    window.Date.prototype.setUTCHours=function(x){return d.setUTCHours(x);}
    window.Date.prototype.setUTCMinutes=function(x){return d.setUTCMinutes(x);}
    window.Date.prototype.setUTCSeconds=function(x){return d.setUTCSeconds(x);}
    window.Date.prototype.setUTCMilliseconds=
        function(x) {return d.setUTCMilliseconds(x);}
  
    window.Date.prototype.toUTCString=function(){return d.toUTCString();}
    window.Date.prototype.toGMTString=function(){return d.toGMTString();}
    window.Date.prototype.toString=function(){return d.toUTCString();}
    window.Date.prototype.toLocaleString=function(){return d.toUTCString();}
    
    /* Fuck 'em if they can't take a joke: */
    window.Date.prototype.toLocaleTimeString=function(){return d.toUTCString();}
    window.Date.prototype.toLocaleDateString=function(){return d.toUTCString();}
    window.Date.prototype.toDateString=function(){return d.toUTCString();}
    window.Date.prototype.toTimeString=function(){return d.toUTCString();}
    
    /* Hack to solve the problem of multiple date objects
     * all sharing the same lexically scoped d every time a new one is
     * created. This hack creates a fresh new prototype reference for 
     * the next object to use with a different d binding.
     * It doesn't break stuff because at the start of this function, 
     * the interpreter grabbed a reference to Date.prototype. During 
     * this function we modified Date.prototype to create the new methods
     * with the lexically scoped d reference.
     */

    // valueOf gets called for implicit string conversion??
    window.Date.prototype = window.eval(window.Object.prototype.toSource());
    return d.toUTCString();
  }

  // Need to do this madness so that we use the window's notion of Function
  // for the constructor. If this is not done, then changes to Function
  // and Object in the real window are not propogated to Date (for example,
  // to extend the Date class with extra functions via a generic inheritance 
  // framework added onto Object - this is done by livejournal and others)
  var newWrappedDate = window.eval("function() { return newDate(); }");

  newWrappedDate.parse=function(s) {
    var d = new origDate(s);
    if(typeof(s) == "string") reparseDate(d, s);
    return d.getTime();    
  }

  newWrappedDate.now=function(){return origDate.now();}
  newWrappedDate.UTC=function(){return origDate.apply(origDate, arguments); }

  // d = new Date();
  // d.__proto__ === Date.prototype
  // d.constructor === Date
  // d.__proto__ === d.constructor.prototype
  // Date.prototype.__proto__  ===  Date.prototype.constructor.prototype 
  // window.__proto__ === Window.prototypea
  
  // XXX: This is still not enough.. But at least we get to claim the 
  // unmasking is violating ECMA-262 by allowing the deletion of var's 
  // (FF Bug 419598)
  with(window) {
    var Date = newWrappedDate;
  }

  } // window.__tb_hook_date == true

  with(window) {
      XPCNativeWrapper = function(a) { return a; };
  }
  with(window.__proto__) {
      XPCNativeWrapper = function(a) { return a; };
  }

  // Gain access to the implict global object (which interestingly claims
  // to be a 'Window' but is not the same class as 'window'...) and 
  // hide XPCNativeWrapper there.
  // This seems no longer necessary in FF2.0.0.13+, and may break FF3?
  //with(window.valueOf.call().__proto__) {
  //    XPCNativeWrapper = function(a) { return a; };
  //}

  // FIXME: Commenting this may open us to unmasking, but allows the hooks to
  // work on FF4. Since we lost the bulletproof resolution masking battle to
  // CSS3 anyways, let's just do what we can.
  //window.__proto__ = null; // Prevent delete from unmasking our properties.
  return true;
}

if (typeof(window.__HookObjects) != "undefined") {
    var res = 23;

    if(!window.__HookObjects()) {
        res = 13;
    }

    window.__HookObjects = undefined;
    delete window['__HookObjects'];
    delete window['__CheckFlag'];
    delete window['__tb_set_uagent'];
    delete window['__tb_oscpu'];
    delete window['__tb_platform'];
    delete window['__tb_productSub'];

    window.__tb_hooks_ran = true;

    res; // Secret result code.
} else {
    42;
}
